### PR TITLE
[PM-7025] include check-run in workflows where secrets are used

### PR DESCRIPTION
**Pull request by bwdil**
## Type of change

GitHub workflows

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ X ] Other
```

## Objective

As part of our organization-wide effort to secure bw code repos, a change to include `check-run` in workflows where `secrets` are used is required.

## Code changes

Incremental change to existing workflows

### CODE REVIEW

File: .github/workflows/build-cli.yml [hunk 1]
```diff
@@ -27,12 +27,18 @@ on:
       - '.github/workflows/build-cli.yml'
   workflow_dispatch:
     inputs: {}
+  pull_request_target:
```
Comment (withinfocus) [inline .github/workflows/build-cli.yml:30]: ❌ You have to replace the above `pull_request_target` with this.

File: .github/workflows/build-cli.yml [hunk 2]
```diff
@@ -362,6 +368,7 @@ jobs:
       - cli
       - cli-windows
       - snap
+      - check-run
```
Comment (withinfocus) [inline .github/workflows/build-cli.yml:371]: ❌ This is expanding the work surface but touches on something Oscar was bringing up -- needing secrets, and when. You'll see below that the secret reference is only relevant when on non-PR builds, so this check isn't even necessary.

**Review by withinfocus — CHANGES_REQUESTED** [review]
This PR wouldn't be necessary as-is; when you look at all the other workflows in this repo it might be. I would like to see one PR for the entire repo, as is relevant.

File: .github/workflows/build-cli.yml (hunk 2 above)
Comment (bwdil) [reply .github/workflows/build-cli.yml:371]: ok, in that case we don't need `check-run` for this one.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -8,16 +8,22 @@ on:
...
 
 jobs:
+  check-run:
+    name: Check PR run
+    uses: bitwarden/gh-actions/.github/workflows/check-run.yml@main
+
   test:
     name: Run tests
     runs-on: ubuntu-22.04
+    needs: check-run
```
Comment (withinfocus) [inline .github/workflows/test.yml:26]: 🎨 Let's also assert some token permissions here.

```suggestion
    needs: check-run
    permissions:
      contents: read
      pull-requests: write
```

**Review by withinfocus — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 3 above)
Comment (bwdil) [reply .github/workflows/test.yml:26]: @withinfocus leaving a reference to [assigning-permissions-to-jobs](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs) 

`contents: read` permits an action to list the commits
`pull-requests: write` permits an action to add a label to a pull request

**Review by withinfocus — APPROVED** [review]

**Review by vgrassia — APPROVED** [review]

**Pull request merged**